opm/grid/cpgrid/CpGridData.cpp

```diff
@@ -512,380 +1199,393 @@ void createInterfaces(std::vector<std::map<int,char> >& attributes,
 
 }
 
-#endif // #if HAVE_MPI
+void CpGridData::computeGeometry(CpGrid& grid,
+                                 const DefaultGeometryPolicy&  globalGeometry,
+                                 const OrientedEntityTable<0, 1>& globalCell2Faces,
+                                 DefaultGeometryPolicy& geometry,
+                                 const OrientedEntityTable<0, 1>& cell2Faces,
+                                 const std::vector< std::array<int,8> >& cell2Points)
+{
+    FaceGeometryHandle faceGeomHandle(globalGeometry.geomVector(std::integral_constant<int,1>()),
+                                      geometry.geomVector(std::integral_constant<int,1>()));
+    FaceViaCellHandleWrapper<FaceGeometryHandle>
+        wrappedFaceGeomHandle(faceGeomHandle, globalCell2Faces, cell2Faces);
+    grid.scatterData(wrappedFaceGeomHandle);
+
+    PointGeometryHandle pointGeomHandle(globalGeometry.geomVector(std::integral_constant<int,3>()),
+                                             geometry.geomVector(std::integral_constant<int,3>()));
+    grid.scatterData(pointGeomHandle);
+
+    CellGeometryHandle cellGeomHandle(globalGeometry.geomVector(std::integral_constant<int,0>()),
+                                      geometry.geomVector(std::integral_constant<int,0>()),
+                                      geometry.geomVector(std::integral_constant<int,3>()),
+                                      cell2Points);
+    grid.scatterData(cellGeomHandle);
+}
 
+void computeFace2Point(CpGrid& grid,
+                       const OrientedEntityTable<0, 1>& globalCell2Faces,
+                       const GlobalIdSet& globalIds,
+                       const OrientedEntityTable<0, 1>& cell2Faces,
+                       const Opm::SparseTable<int>& globalFace2Points,
+                       Opm::SparseTable<int>& face2Points,
+                       const std::map<int,int>& global2local,
+                       std::size_t noFaces)
+{
+    std::vector<int> rowSizes(noFaces);
+    using EntityTable = SparseTableEntity<1>;
+    using RowSizeDataHandle = RowSizeDataHandle<EntityTable, 1>;
+    EntityTable wrappedGlobal(globalFace2Points);
+    RowSizeDataHandle rowSizeHandle(wrappedGlobal, rowSizes);
+    FaceViaCellHandleWrapper<RowSizeDataHandle>
+        wrappedSizeHandle(rowSizeHandle, globalCell2Faces, cell2Faces);
+    grid.scatterData(wrappedSizeHandle);
+    face2Points.allocate(rowSizes.begin(), rowSizes.end());
+    // Use entity with index INT_MAX to mark unprocessed row entries
+    for (int row = 0, size = face2Points.size(); row < size; ++row)
```

Comment:
range for?

Reply:
Probably did not send my comment. SparseTable provides no row iterators and I did not want to change that. Hence no.

Reply:
I'll add those iterators to SparseTable (but not right now).